SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -937,6 +937,23 @@ - (OWSMessageSend *)getSessionRestoreMessageForHexEncodedPublicKey:(NSString *)h
     return [[OWSMessageSend alloc] initWithMessage:message thread:thread recipient:recipient senderCertificate:senderCertificate udAccess:theirUDAccess localNumber:userHexEncodedPublicKey success:^{ } failure:^(NSError *error) { }];
 }
 
+- (LKFriendRequestMessage *)getMultiDeviceFriendRequestMessageForHexEncodedPublicKey:(NSString *)hexEncodedPublicKey inThread:(TSContactThread *)thread transaction:(YapDatabaseReadWriteTransaction *)transaction
```

Comment:
This is duplicating a fair bit of code

Reply:
Hmm...This should be my dirty solution for the yapdatabase transaction issue. I was facing a transaction in transaction issue...